src/coreclr/vm/dllimport.cpp

```diff
@@ -662,6 +662,12 @@ class ILStubState : public StubState
         }
 #endif // PROFILING_SUPPORTED
 
+        if (SF_IsCheckPendingException(m_dwStubFlags)
+            && SF_IsForwardStub(m_dwStubFlags))
+        {
+            pcsDispatch->EmitCALL(METHOD__STUBHELPERS__CLEAR_PENDING_EXCEPTION_OBJECT, 0, 0);
```

Comment:
Do we really need this? I am not sure how important it is to try to be robust against somebody setting the pending by accident without a matching for pending exception.

Reply:
I feel better with it because it provides us with a guarantee that nothing was missed. I assume your concern is the unnecessary performance impact?

Reply:
I agree that it can cover up some a problem in simple cases. It does not provide the guarantee when the calls are nested. If there is a real problem that this is covering up, it will be hard to repro and debug.

I am not that concerned about performance. I have been more wondering about what made you to add this.

Reply:
> I have been more wondering about what made you to add this.

It was the potential stale thread state that bothered me. We have a clear/get/set for the last error so figured I would mimic the behavior. It was also when it was possible to get into a state where we didn't check, but that should be handled now. If you don't see value I am fine removing it.

Reply:
I would remove it until we have a proof that it is really needed.